play-json/jvm/src/test/scala/play/api/libs/json/ReadsSpec.scala

```diff
@@ -745,6 +743,75 @@ class ReadsSpec extends org.specs2.mutable.Specification {
     }
   }
 
+  "BigDecimal numbers" should {
+    val DefaultReads = implicitly[Reads[BigDecimal]]
+    import DefaultReads.reads
+
+    val settings = JsonParserSettings.settings.bigDecimalParseSettings
+
+    val longNumberString =
+      Iterator
+        .fill(settings.digitsLimit)("1")
+        .mkString
+
+    "parse long string" in {
+      val input = JsString(longNumberString)
+      reads(input)
+        .must(beLike {
+          case JsSuccess(_, _) => ok
+        })
+    }
+
+    "not parse string exceeding length limit" in {
+      val input = JsString(longNumberString + "1")
+      reads(input).must(beLike {
+        case JsError((_, JsonValidationError("error.expected.numberdigitlimit" :: Nil) :: Nil) :: Nil) => ok
+      })
+    }
+
+    "parse string with acceptable scale" in {
+      val numberString = s"1E+${settings.scaleLimit}"
+      reads(JsString(numberString)).must_===(JsSuccess(BigDecimal(numberString)))
+    }
+
+    "not parse string exceeding scale limit" in {
+      val numberString = s"1E+${settings.scaleLimit + 1}"
+      reads(JsString(numberString)).must(beLike {
+        case JsError((_, JsonValidationError("error.expected.numberscalelimit" :: Nil) :: Nil) :: Nil) => ok
+      })
+    }
+  }
+
+  "BigDecimal numbers" should {
+    val DefaultReads = implicitly[Reads[BigInt]]
+    import DefaultReads.reads
+
+    val settings = JsonParserSettings.settings.bigDecimalParseSettings
+
+    val longNumberString =
+      Iterator
+        .fill(settings.digitsLimit)("1")
+        .mkString
+
+    "parse long string" in {
+      val input = JsString(longNumberString)
+      reads(input)
```

Comment:
Avoid unnecessary paren nesting
```suggestion
      reads(input) must beLike {
```